src/clients/c++/request_http.cc

```diff
@@ -1159,9 +1160,15 @@ HttpRequestImpl::CreateResult(
   }
 
   std::unique_ptr<ResultImpl> result(new ResultImpl(infer_output, batch_size));
-  result->SetBatch1Shape(output.raw().dims());
-  if (IsFixedSizeDataType(infer_output->DType())) {
-    result->SetBatchnByteSize(output.raw().batch_byte_size());
+
+  if (!ctx.UsesSharedMemory(output.name())) {
+    result->SetBatch1Shape(output.raw().dims());
```

Comment:
We need to set the shape (and batch-n size??) even if using shared memory... why can't we? How will user know the shape of an variable-sized output if we don't set this?

Reply:
Fixed